Tests/scripts/update_id_set.py

```diff
@@ -396,6 +460,76 @@ def re_create_id_set():
     exec_time = time.time() - start_time
     print_color("Finished the creation of the id_set. Total time: {} seconds".format(exec_time), LOG_COLORS.GREEN)
 
+    duplicates = find_duplicates(new_ids_dict)
+    if duplicates:
+        print_error('The following duplicates were found: {}'.format(duplicates))
+        sys.exit(1)
+
+
+def find_duplicates(id_set):
+    scripts = id_set['scripts']
+    script_ids = set([script.keys()[0] for script in scripts])
+
+    scripts_list = []
+    for script_id in script_ids:
+        if has_duplicate(scripts, script_id):
+            scripts_list.append(script_id)
+
+    integrations = id_set['integrations']
+    integration_ids = set([integration.keys()[0] for integration in integrations])
+
+    integration_list = []
+    for integration_id in integration_ids:
+        if has_duplicate(integrations, integration_id):
+            integration_list.append(integration_id)
+
+    playbooks = id_set['playbooks']
+    playbook_ids = set([playbook.keys()[0] for playbook in playbooks])
+
+    playbooks_list = []
+    for playbook_id in playbook_ids:
+        if has_duplicate(playbooks, playbook_id):
+            integration_list.append(playbook_id)
+
+    test_playbooks = id_set['TestPlaybooks']
+    test_playbook_ids = set([test_playbook.keys()[0] for test_playbook in test_playbooks])
+
+    test_playbooks_list = []
+    for test_playbook_id in test_playbook_ids:
+        if has_duplicate(test_playbooks, test_playbook_id):
+            test_playbooks_list.append(test_playbook_id)
+
+    return scripts_list, integration_list, playbooks_list, test_playbooks_list
+
+
+def has_duplicate(id_set, id_to_check):
+    duplicates = [duplicate for duplicate in id_set if duplicate.get(id_to_check)]
+
+    if len(duplicates) < 2:
+        return False
+
+    for dup1, dup2 in itertools.combinations(duplicates, 2):
+        dict1 = dup1.values()[0]
+        dict2 = dup2.values()[0]
+        dict1_from_version = LooseVersion(dict1.get('fromversion', '0.0.0'))
+        dict2_from_version = LooseVersion(dict2.get('fromversion', '0.0.0'))
+        dict1_to_version = LooseVersion(dict1.get('toversion', '99.99.99'))
+        dict2_to_version = LooseVersion(dict2.get('toversion', '99.99.99'))
+
+        if dict1['name'] != dict2['name']:
+            print_error('The following objects has the same ID but different names: {}, {}.'.format(dict1['name'],
+                                                                                                    dict2['name']))
+        is_duplicate = True
+        if dict1_from_version >= dict2_to_version:
+            is_duplicate = False
+        if dict2_to_version > dict1_to_version:
+            is_duplicate = False
+        if dict2_from_version > dict1_to_version:
+            is_duplicate = False
+        if dict1_from_version > dict2_from_version:
+            is_duplicate = False
+    return is_duplicate
```

Comment:
when you encounter a problem, you should exit immediately - no reason to continue the iterations.
look at my original comment (previous PR):
if (A, C) is the first pair, the other pairs will override the value of `is_duplicate` and the function will return false.